glslc/src/dependency_info.h

```diff
@@ -0,0 +1,163 @@
+  }
+
+  // Checks if the handler's dumping mode is set to dump dependency info as
+  // extra dependency info files.
+  bool DumpingToExtraDependencyInfoFiles() {
+    return mode_ == dump_as_extra_file;
+  }
+
+  // Checks if the handler's dumping mode is set to dump dependency info as
+  // normal compilation output.
+  bool DumpingAsCompilationOutput() {
+    return mode_ == dump_as_compilation_output;
+  }
+
+  // Checks if the handler's dumping mode is not set.
+  bool DumpingModeNotSet() { return mode_ == not_set; }
+
+  // Checks if the handler is at valid state for dumping dependency info.
+  bool IsValid(std::string* error_msg_ptr, size_t num_files) {
+    if (DumpingModeNotSet()) {
+      *error_msg_ptr =
+          "to generate dependencies you must specify either -M (-MM) or -MD";
+      return false;
+    }
+
+    if (!user_specified_dep_file_name_.empty() ||
+        !user_specified_dep_target_label_.empty()) {
+      if (num_files > 1) {
+        *error_msg_ptr =
+            "to specify dependency info file name or dependency info target, "
+            "only one input file is allowed.";
+        return false;
+      }
+    }
+    return true;
+  }
+
+ private:
+  typedef enum {
+    // By default, dump mode should be set to not_set, which means the
+    // dependency handler is instantiated but dumping mode is not set. This
+    // is the case when dependency info file or the target name of the
+    // dependency info is specified by the user (by -MF or MT), but dumping
+    // mode is not set (as -MD or -M is not specified). Compiler should exit
+    // with an error message in such a case.
+    not_set = 0,
+    // Dumping dependency info as normal compilation output mode.
+    // In this mode, the output file name specified by -o or -MF option will
+    // be used for dependency info file, and dependency info will be dumped to
+    // stdout if the user does not specify an output name. Compiler should be
+    // set to preprocessing-only mode, and compilation output won't be generated
+    // in this mode.
+    dump_as_compilation_output,
+    // Dumping dependency info as extra dependency info files mode. In this
+    // mode, dependency info will be dumped to user specified dependency
+    // info file or a *.d file whose name is deduced from main output file
+    // name or input file name. Compilation output will still be generated
+    // along with the dependency info.
+    dump_as_extra_file,
+  } dump_mode;
+
+  // Deduce the target file label to be used in depdendency info file.
+  // If -MT defined a label, use that string as the label. Otherwise,
+  // if compilation_output_file_name is empty, "-" or "a.spv", the label
+  // should be deduced from source_file_name, otherwise, it should be
+  // compilation_output_file_name. If using the source_file_name as the
+  // label but source_file_name is an absolute path, we need to strip the
+  // directory name for the label.
+  std::string GetFinalTargetLabel(
```

Comment:
Can just be GetTarget  ?
The "Final" is just misleading.  The "Label" seems redundant.